pkg/controllers/ocp-controllers.go

```diff
@@ -38,80 +37,98 @@ import (
 	"github.com/openshift/microshift/pkg/util"
 )
 
-func newOpenshiftApiServerCommand(stopCh <-chan struct{}) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "openshift-apiserver",
-		Short: "Command for the OpenShift API Server",
-		Run: func(cmd *cobra.Command, args []string) {
-			cmd.Help()
-			os.Exit(1)
-		},
-	}
-	start := openshift_apiserver.NewOpenShiftAPIServerCommand("start", os.Stdout, os.Stderr, stopCh)
-	cmd.AddCommand(start)
+type OCPAPIServer struct {
+	ConfigFile string
+	Output     io.Writer
+	cfg        *config.MicroshiftConfig
+}
 
-	return cmd
+func NewOpenShiftAPIServer(cfg *config.MicroshiftConfig) *OCPAPIServer {
+	s := &OCPAPIServer{}
+	s.configure(cfg)
+	return s
 }
 
-func OCPAPIServer(cfg *config.MicroshiftConfig) error {
-	stopCh := make(chan struct{})
-	command := newOpenshiftApiServerCommand(stopCh)
+func (s *OCPAPIServer) Name() string           { return "ocp-apiserver" }
+func (s *OCPAPIServer) Dependencies() []string { return []string{"kube-apiserver"} }
+
+func (s *OCPAPIServer) configure(cfg *config.MicroshiftConfig) error {
+	var configFilePath = cfg.DataDir + "/resources/openshift-apiserver/config/config.yaml"
+
+	if err := config.OpenShiftAPIServerConfig(cfg); err != nil {
+		logrus.Infof("Failed to create a new ocp-apiserver configuration: %v", err)
+		return err
+	}
 	args := []string{
-		"start",
```

Comment:
what happen to these args?

Reply:
They are read from the config file when `RunAPIServer` is called  
[here](https://github.com/redhat-et/microshift/blob/e9c795bf061a2bc7dce47adc03e1965c2281bfe0/pkg/controllers/ocp-controllers.go#L108)